arm-storsimple8000series/2017-06-01/swagger/storsimple.json

```diff
@@ -0,0 +1,1778 @@
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/managerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The encryption settings.",
+            "schema": {
+              "$ref": "#/definitions/EncryptionSettings"
+            }
+          }
+        },
+        "deprecated": false
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorSimple/managers/{managerName}/extendedInformation/vaultExtendedInfo": {
+      "get": {
+        "tags": [
+          "Managers"
+        ],
+        "description": "Returns the extended information of the specified manager name.",
+        "operationId": "Managers_GetExtendedInfo",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/managerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The manager extended information.",
+            "schema": {
+              "$ref": "#/definitions/ManagerExtendedInfo"
+            }
+          }
+        },
+        "deprecated": false
+      },
+      "put": {
+        "tags": [
+          "Managers"
+        ],
+        "description": "Creates the extended info of the manager.",
+        "operationId": "Managers_CreateExtendedInfo",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "The manager extended information.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ManagerExtendedInfo"
+            }
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/managerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The extended information on the manager has been successfully created",
+            "schema": {
+              "$ref": "#/definitions/ManagerExtendedInfo"
```

Comment:
@ravbhatnagar, in our case, only 'Manager' is the ARM-Resource(billable entity), and hence we have derived it from 'Resource' Model. All other Models(like Devices, ManagerExtendedInfo, etc
for now, and  in later PRs we will see BandwidthSettings, Volumes, VolumeContainers, etc) are non-ARM Resources and hence are derived from 'BaseModel'. And in both 'Resource' and 'BaseModel' we have the common properties 'id', 'name' and 'type'.

But in 'Resource' Model, we have extra properties: 'location' and 'tags' which are specifically required for ARM-Resource.

Hence, should these extra properties be used to decide if it's a resource or not?